sympy/sets/sets.py

```diff
@@ -1202,28 +1202,32 @@ def reduce(args):
 
         # ===== Pair-wise Rules =====
         # Here we depend on rules built into the constituent sets
-        args = set(args)
+        args = FiniteSet(*args)
```

Comment:
few months before I came across similar problem ([comparison error](https://github.com/sympy/sympy/pull/10733#issuecomment-216266502)), but that looks different case.